### ISSUE TITLE
java.lang.ArrayIndexOutOfBoundsException

### ISSUE THREAD

**Issue by AxeOfMen** [marcello-dev/java-call-graph-plotter#5]
I got an ArrayIndexOutOfBoundsException when I try to run the program:

```
Exception in thread "main" java.lang.ArrayIndexOutOfBoundsException: Index 1 out of bounds for length 1
	at gr.gousiosg.javacg.stat.DynamicCallManager.retrieveCalls(DynamicCallManager.java:84)
	at gr.gousiosg.javacg.stat.ClassVisitor.visitJavaClass(ClassVisitor.java:65)
	at gr.gousiosg.javacg.stat.ClassVisitor.start(ClassVisitor.java:92)
	at gr.gousiosg.javacg.stat.JCallGraph.lambda$1(JCallGraph.java:78)
	at java.base/java.util.stream.ReferencePipeline$7$1.accept(ReferencePipeline.java:271)
	at java.base/java.util.Iterator.forEachRemaining(Iterator.java:133)
	at java.base/java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1801)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:913)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.reduce(ReferencePipeline.java:563)
	at gr.gousiosg.javacg.stat.JCallGraph.main(JCallGraph.java:81)
```

**Comment by marcello-dev** [comment]
The error seems to come from another library. Did you run with the example jar or your own jar?

**Comment by marcello-dev** [comment]
Closed because of no activity

**Comment by a-t-0** [comment]
Thank you for your work on creating a visualisation of the call graphs in java, and for including such a clear Readme.md.
I got a similar error on:
```sh
java -jar javacg/javacg-0.1-SNAPSHOT-static.jar /home/name/git/callgraphs/some.jar > call-graph.txt
Exception in thread "main" java.lang.ArrayIndexOutOfBoundsException: Index 1 out of bounds for length 1
	at gr.gousiosg.javacg.stat.DynamicCallManager.retrieveCalls(DynamicCallManager.java:84)
	at gr.gousiosg.javacg.stat.ClassVisitor.visitJavaClass(ClassVisitor.java:65)
	at gr.gousiosg.javacg.stat.ClassVisitor.start(ClassVisitor.java:92)
	at gr.gousiosg.javacg.stat.JCallGraph.lambda$1(JCallGraph.java:78)
	at java.base/java.util.stream.ReferencePipeline$7$1.accept(ReferencePipeline.java:273)
	at java.base/java.util.Iterator.forEachRemaining(Iterator.java:133)
	at java.base/java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1921)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:921)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.reduce(ReferencePipeline.java:667)
	at gr.gousiosg.javacg.stat.JCallGraph.main(JCallGraph.java:81)
```

Based on your comment, I assumed it was because I did not pass the required arguments into my own `some.jar` (which throws an error when it does not receive any arugments). However, the error that `some.jar` throws (upon missing arguments) is:
```sh
Exception in thread "main" com.someproject.some.CommandLineParseException: Error parsing command-line arguments
	at com.someproject.some.SomeParser.parsesomeArgs(SomeParser.java:424)
	at com.someproject.some.SomeParser.<init>(SomeParser.java:40)
	at com.someproject.Main.main(Main.java:38)
Caused by: org.apache.commons.some.MissingOptionException: Missing required options: configfilepath
	at org.apache.commons.some.DefaultParser.checkRequiredOptions(DefaultParser.java:243)
	at org.apache.commons.some.DefaultParser.parse(DefaultParser.java:722)
	at org.apache.commons.some.DefaultParser.parse(DefaultParser.java:679)
	at org.apache.commons.some.DefaultParser.parse(DefaultParser.java:660)
	at com.someproject.some.SomeParser.parsesomeArgs(SomeParser.java:418)
```

It could be however, that your code throws the indexoutofbounds error, if the `.jar` file it runs on, throws an error. 

0. Would you perhaps be willing and able to shine some light on this matter? 
1. Would it for example be possible to just paste the arguments for our own jar, at the end of your command to make it work (such that those args are passed on) (I tried this but the error remained the same, so perhaps I was doing that wrong)?

**Comment by marcello-dev** [comment]
I could replicate the issue with another jar, the error comes from the dependent project but I'll look into it.

FYI The example jar in the folder target-jar still works, if you want to have a look at the source code https://github.com/marcello-dev/demo-customer

**Comment by hjake123** [comment]
I have also experienced this error, my guess given the behavior of it is that the program has issues when the provided .jar file doesn't have a main method. I was able to successfully parse both the example jar and a few random java applications, but trying to parse a Minecraft mod (a jar file that has no main) caused the same crash experienced above. 

I was able to reproduce with various mods, including the latest version of [Just Enough Items](https://legacy.curseforge.com/minecraft/mc-mods/jei)

Crash log:
```
Exception in thread "main" java.lang.ArrayIndexOutOfBoundsException: Index 1 out of bounds for length 1
        at gr.gousiosg.javacg.stat.DynamicCallManager.retrieveCalls(DynamicCallManager.java:84)
        at gr.gousiosg.javacg.stat.ClassVisitor.visitJavaClass(ClassVisitor.java:65)
        at gr.gousiosg.javacg.stat.ClassVisitor.start(ClassVisitor.java:92)
        at gr.gousiosg.javacg.stat.JCallGraph.lambda$1(JCallGraph.java:78)
        at java.base/java.util.stream.ReferencePipeline$7$1.accept(ReferencePipeline.java:273)
        at java.base/java.util.Iterator.forEachRemaining(Iterator.java:133)
        at java.base/java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1845)
        at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
        at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
        at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:921)
        at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
        at java.base/java.util.stream.ReferencePipeline.reduce(ReferencePipeline.java:667)
        at gr.gousiosg.javacg.stat.JCallGraph.main(JCallGraph.java:81)
```

**Comment by marcello-dev** [comment]
Fixed it, please check by re-running the command to generate the call graph.

**Comment by a-t-0** [comment]
![image](https://github.com/marcello-dev/java-call-graph-plotter/assets/34750068/3e486320-9be2-49dd-a258-342c665499d6)

Thank you! I confirm the error is resolved. For my .jar file, it yields an empty image, but for your example command, it does show the graph:
![image](https://github.com/marcello-dev/java-call-graph-plotter/assets/34750068/94ac37d8-d259-4136-95a9-a8d65625fbd4)
I have not yet figured out whether that is because my jar misses the CLI args, or because it is just weird or perhaps too large or something else.However, that is another issue.

**Comment by marcello-dev** [comment]
Hi @a-t-0 have you modified the node selector for your use case? For example, implementing [this class](https://github.com/marcello-dev/java-call-graph-plotter/blob/master/nodeselectors.py#L84-L97) and then using it when running the python script

**Comment by a-t-0** [comment]
A short answer is initially I did not, after your feedback I did. Thank you. If I recall correctly, I noticed that the text file that is created by the plotter was over 295 mb large, and it hangs indefinitely when I try to convert that textfile to a graph. I did notice that your code worked on the example case you provided. So I think I should look for a different solution, though that does not enjoy my current priority.
